api/src/db/models/opportunity_models.py

```diff
@@ -435,5 +435,7 @@ class OpportunitySearchIndexQueue(ApiSchemaTable, TimestampMixin):
     opportunity_id: Mapped[int] = mapped_column(
         BigInteger, ForeignKey(Opportunity.opportunity_id), primary_key=True, index=True
     )
-    opportunity: Mapped[Opportunity] = relationship(Opportunity)
+    opportunity: Mapped[Opportunity] = relationship(
+        Opportunity, cascade="all, delete-orphan", single_parent=True
+    )
```

Comment:
You would need this relationship to go the other way. Doing this would delete the opportunity if you deleted the queue record.

Reply:
Ah, good call. I thought the `single_parent` property would have done this on the queue. Updated this.